### PR TITLE
Replace setup.py with pyproject.toml

### PR DESCRIPTION
**Pull request by tamarl08**
- [x] TODO: update documentation

See the package in [testPyPI](https://test.pypi.org/project/skll/5.0.2/) (I incremented the version just for the test)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,68 @@
...
+filter_features = "skll.utils.commandline.filter_features:main"
+generate_predictions = "skll.utils.commandline.generate_predictions:main"
+join_features = "skll.utils.commandline.join_features:main"
+print_model_weights = "skll.utils.commandline.print_model_weights:main"
+run_experiment = "skll.utils.commandline.run_experiment:main"
+skll_convert = "skll.utils.commandline.skll_convert:main"
+summarize_results = "skll.utils.commandline.summarize_results:main"
+compute_eval_from_predictions = "skll.utils.commandline.compute_eval_from_predictions:main"
+plot_learning_curves = "skll.utils.commandline.plot_learning_curves:main"
+
```
Comment (desilinguist) [inline pyproject.toml:56]:
```
[tool.setuptools]
packages = ["skll"]
```

**Review by desilinguist — CHANGES_REQUESTED** [review]

**Comment by desilinguist** [comment]
I tested both `pip install -e .` and `python -m build` and both work fine. I suggested to add a section since I had it in my other `pyproject.toml` files and figured it can't hurt.

File: pyproject.toml (hunk 1 above)
Comment (tamarl08) [reply pyproject.toml:56]: isn't it redundance to have it with the ```[tool.setuptools.packages.find]``` part?
[This](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#custom-discovery) is the documentation I'm looking at.
Maybe I shouldn't set `namespace` to false?

Comment (desilinguist) [reply pyproject.toml:56]: Yeah looks like `python -m build` finds all the correct files so may be we don't need this. Can you try setting up an environment with the new package from TestPyPI to make sure it works?

Comment (tamarl08) [reply pyproject.toml:56]: that's what I did. you mean with this change?

**Review by desilinguist — APPROVED** [review]

**Review by damien2012eng — APPROVED** [review]

**Pull request merged**